### PR TITLE
chore: split env config and add util, add commitlint, add pr template…

### PR DESCRIPTION
**Pull request by JonLazariniDD**
…, set up hooks, add project config/metadata

(cherry picked from commit a2ac7d88b0e99b1ecb5500642cf108775df1c6ad)

chore: fix linting issues

(cherry picked from commit 60695c515fa218c3858fc13d1003d6d7eb9f6806)

chore: remove console.log

### CODE REVIEW

File: src/app/app.config.ts [hunk 1]
```diff
@@ -1,30 +1,20 @@
...
-          if (!(environment.name === 'deploy')) {
-            this._settings = <IAppConfig>configJsonDev.default;
-          } else {
-            this._settings = <IAppConfig>configJsonDeploy.default;
-          }
-          console.log(`adalConfig => ${JSON.stringify(this._settings.adalConfig)}`);
-        })
-      )
-      .toPromise();
+  load(): Promise<IAppConfig> {
```
Comment (JonLazariniDD) [inline src/app/app.config.ts:13]: Will need to use `environments/` instead in the future and require a ticket for it I think. This was in place prior to ADR again.

File: package.json [hunk 2]
```diff
@@ -5,19 +5,24 @@
...
     "test": "jest --coverage",
     "test:dev": "jest --watch",
     "lint": "ng lint",
+    "lint:fix": "ng lint --fix",
     "e2e": "cypress open",
     "sonar-scanner": "npm run test && sonar-scanner",
     "test:deploy": "npm run build && npx http-server ./dist -p 4200 -c-1",
-    "prepush": "npm run build && npm test",
-    "tslint": "tslint src/**/*.ts",
+    "prepush": "npm test",
```
Comment (alleypa) [inline package.json:25]: I will suggest we still build locally during prepush. Sometimes the templates are broken during development and you do get the quick feedback of the broken template(s) during build. Though building locally is very slow. We can investigate why separately

File: src/app/directives/form-connector/form-connector.directive.ts [hunk 3]
```diff
@@ -21,14 +21,9 @@ export class FormConnectorDirective implements OnInit, OnDestroy {
...
       .pipe(
-        debounceTime(this.debounce),
-        tap((value) => {
-          console.log('VALUE:', value, 'PATH:', this.path);
-        })
+        debounceTime(this.debounce)
       )
       .subscribe();
-
-    console.log(this.formGroupDirective.form);
```
Comment (alleypa) [inline src/app/directives/form-connector/form-connector.directive.ts:31]: @JonLazariniDD  this directive was written mainly as diagnosis tool. It helps in understanding how the parent formGroup behaves down to the nth component. Once all the big tickets are done and settled the directive job will be done. Please lets leave for now and remove later

File: .secignore [hunk 4]
```diff
@@ -0,0 +1 @@
+.npmrc
```
Comment (m-ujaffer) [inline .secignore:1]: line break

File: .npmrc [hunk 5]
```diff
@@ -0,0 +1,2 @@
+audit-level = high
+audit = true
```
Comment (m-ujaffer) [inline .npmrc:2]: line break

File: .gitignore [hunk 6]
```diff
@@ -43,3 +43,6 @@ Thumbs.db
 .sonar
 .scannerwork
 .reports
+
+*.env
+/**/*.dev.json
```
Comment (m-ujaffer) [inline .gitignore:48]: line break

**Pull request merged**